### PR TITLE
Add helper tools for integration development in the Engine

### PR DESCRIPTION
**Pull request by JavierBejMen**
|Related issue|
|---|
|#24046|

This PR adds the following helper python scripts:
- flatt output:
```
$ python3 tools/flatt_output.py -h
usage: flatt_output.py [-h] [-r ROOT] path pattern

Flattens the output json of a root field

positional arguments:
  path                  Path to the input directory
  pattern               Pattern for the output files

options:
  -h, --help            show this help message and exit
  -r ROOT, --root ROOT  Root field to flatten. Default fllattens all fields
```

- update expected:
```
$ python3 tools/update_expected.py -h
usage: update_expected.py [-h] [-o OUTPUT] [-b BINARY] environment path pattern test_integration test_integration_conf_file

Update expected results

positional arguments:
  environment           Environment folder where the test configuration is stored.
  path                  Path to the directory with input files
  pattern               Pattern for the input files
  test_integration      engine-test integration if any
  test_integration_conf_file
                        engine-test integration configuration file

options:
  -h, --help            show this help message and exit
  -o OUTPUT, --output OUTPUT
                        Path to the output file, default: /tmp/update_expected/
  -b BINARY, --binary BINARY
                        Specify the path to the engine binary
```

Also QoL changes on:
- src/engine/test/health_test/initialState.py
- src/engine/test/setupEnvironment.py
- src/engine/tools/compare_expected.py

### CODE REVIEW

File: src/engine/tools/flatt_output.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+from pathlib import Path
```
Comment (juliancnn) [inline src/engine/tools/flatt_output.py:1]:
```suggestion
#!/usr/bin/env python3

from pathlib import Path
```

File: src/engine/tools/update_expected.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
```
Comment (juliancnn) [inline src/engine/tools/update_expected.py:1]: #!/usr/bin/env python3
```suggestion
#!/usr/bin/env python3

from pathlib import Path
```

**Review by juliancnn — CHANGES_REQUESTED** [review]
The file `src/engine/tools/flatt_output.py` should be renamed (double `t` in the name):
 `src/engine/tools/flat_output.py`  

On the other hand, it would be desirable to add execution permissions to the scripts:
```
chmod +x tools/flatt_output.py 
chmod +x tools/update_expected.py 
chmod +x tools/compare_expected.py 
```

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**